tests/base.py

```diff
@@ -133,15 +133,14 @@ def _add_gids_per_port(self, ctx, dev, port):
         # Don't add ports which are not active
         if ctx.query_port(port).state != e.IBV_PORT_ACTIVE:
             return
-        idx = 0
-        while True:
+
+        for idx in range(ctx.query_port(port).gid_tbl_len):
             gid = ctx.query_gid(port, idx)
             if gid.gid[-19:] == self.ZERO_GID:
```

Comment:
I'm not sure this existing logic is correct, @paravmellanox ? Can we have ZERO_GID holes in the gid table? I think so?

Reply:
@jgunthorpe 
@Kamalheib 
yes, gid table is constructed out of IPv4 and IPv6 addresses for physical port, vlan, macvlan, bond device etc for RoCE.

and IP addresses they can be added and deleted in any sequence. with DHCP also lease can expire, resulting into IP address removal and hence GID removal, that leads to holes.
so it should iterate over whole GID table.

Reply:
Thanks for reviewing the patch.
I'll update the patch to iterate over the whole GID table.